tempesta_fw/t/functional/helpers/deproxy.py

```diff
@@ -491,6 +532,8 @@ def handle_read(self):
                            '<<<<<\n%s>>>>>'
                            % self.response_buffer))
             raise
+        if len(self.response_buffer) > 0:
+            raise ParseError('Garbage after response end:\n```\n%s\n```\n' % self.response_buffer)
```

Comment:
Ok, now it's makes the job. But a either a `TODO` comment or rework is required here. Currently both client and response don't support message pipelining because of tests structure: messages from TempestaFW are validated one-by one (The test suite was developed before pipelining was supported). We may configure Client client with the full chain list and give it some instructions about when to use the pipelining. When the Server receives a new message it can save it to the right slot in chain list. Not all tests allow pipelining.

Reply:
Now we don't have pipelined functional test. So I've added  "TODO"